packages/components/src/DataList/components/DataListHeaderTile/DataListHeaderTile.tsx

```diff
@@ -0,0 +1,72 @@
+import React from "react";
+import classnames from "classnames";
+import styles from "./DataListHeaderTile.css";
+import { DataListHeader, DataListObject } from "../../DataList.types";
+import { Text } from "../../../Text";
+import { Icon } from "../../../Icon";
+import { useDataListContext } from "../../context/DataListContext";
+
+interface DataListHeaderTileProps<T extends DataListObject> {
+  headers: DataListHeader<T>;
+  headerKey: string;
+}
+
+export function DataListHeaderTile<T extends DataListObject>({
+  headers,
+  headerKey,
+}: DataListHeaderTileProps<T>) {
+  const { sorting } = useDataListContext();
+
+  const isSortable = sorting?.sortable.includes(headerKey);
+  const sortingState = sorting?.state;
+
+  function toggleSorting(sortingKey: string) {
+    if (sortingKey === sortingState?.key) {
+      sorting?.onSort({
+        direction: sortingState.direction === "asc" ? "desc" : "asc",
+        key: sortingKey,
+      });
+    } else {
+      sorting?.onSort({ direction: "asc", key: sortingKey });
+    }
+  }
+
+  return (
+    <div
```

Comment:
shouldn't use div on anything interactable as this would only work for mouse and tap but not accessibility

Reply:
What element could be usable here? Would a `span` be accessible in this case?

Reply:
Added a button to sortable columns